source/DataElement/AMDSFlatArray.cpp

```diff
@@ -198,3 +199,437 @@ AMDSFlatArray AMDSFlatArray::operator /(const quint32 divisor)
+	case AMDSDataTypeDefinitions::Signed8:
+		if(vectorQint8_.size() != targetArray->vectorQint8().size())
+			return false;
+		memcpy(targetArray->vectorQint8().data(), vectorQint8_.constData(), vectorQint8_.size()*sizeof(qint8));
+		return true;
+	case AMDSDataTypeDefinitions::Unsigned8:
+		if(vectorQuint8_.size() != targetArray->vectorQuint8().size())
+			return false;
+		memcpy(targetArray->vectorQuint8().data(), vectorQuint8_.constData(), vectorQuint8_.size()*sizeof(quint8));
+		return true;
+	case AMDSDataTypeDefinitions::Signed16:
+		if(vectorQint16_.size() != targetArray->vectorQint16().size())
+			return false;
+		memcpy(targetArray->vectorQint16().data(), vectorQint16_.constData(), vectorQint16_.size()*sizeof(qint16));
+		return true;
+	case AMDSDataTypeDefinitions::Unsigned16:
+		if(vectorQuint16_.size() != targetArray->vectorQuint16().size())
+			return false;
+		memcpy(targetArray->vectorQuint16().data(), vectorQuint16_.constData(), vectorQuint16_.size()*sizeof(quint16));
+		return true;
+	case AMDSDataTypeDefinitions::Signed32:
+		if(vectorQint32_.size() != targetArray->vectorQint32().size())
+			return false;
+		memcpy(targetArray->vectorQint32().data(), vectorQint32_.constData(), vectorQint32_.size()*sizeof(qint32));
+		return true;
+	case AMDSDataTypeDefinitions::Unsigned32:
+		if(vectorQuint32_.size() != targetArray->vectorQuint32().size())
+			return false;
+		memcpy(targetArray->vectorQuint32().data(), vectorQuint32_.constData(), vectorQuint32_.size()*sizeof(quint32));
+		return true;
+	case AMDSDataTypeDefinitions::Signed64:
+		if(vectorQint64_.size() != targetArray->vectorQint64().size())
+			return false;
+		memcpy(targetArray->vectorQint64().data(), vectorQint64_.constData(), vectorQint64_.size()*sizeof(qint64));
+		return true;
+	case AMDSDataTypeDefinitions::Unsigned64:
+		if(vectorQuint64_.size() != targetArray->vectorQuint64().size())
+			return false;
+		memcpy(targetArray->vectorQuint64().data(), vectorQuint64_.constData(), vectorQuint64_.size()*sizeof(quint64));
+		return true;
+	case AMDSDataTypeDefinitions::Float:
+		if(vectorFloat_.size() != targetArray->vectorFloat().size())
+			return false;
+		memcpy(targetArray->vectorFloat().data(), vectorFloat_.constData(), vectorFloat_.size()*sizeof(float));
+		return true;
+	case AMDSDataTypeDefinitions::Double:
+		if(vectorDouble_.size() != targetArray->vectorDouble().size())
+			return false;
+		memcpy(targetArray->vectorDouble().data(), vectorDouble_.constData(), vectorDouble_.size()*sizeof(double));
+		return true;
+	case AMDSDataTypeDefinitions::InvalidType:
+		return -1;
+	default:
+		return -1;
+	}
+}
+
+bool AMDSFlatArray::resetTargetArrayAndReplaceData(AMDSFlatArray *targetArray) const{
+	targetArray->clearAndReset(dataType_, size());
+	return copyDataToTargetArray(targetArray);
+}
+
+void AMDSFlatArray::resizeType(AMDSDataTypeDefinitions::DataType dataType, quint32 size){
+	dataType_ = dataType;
+
+	switch(dataType_){
+	case AMDSDataTypeDefinitions::Signed8:
```

Comment:
After resizing, are the contents of of the array still valid?  Should they be?  If not, you could get the resizing and initializing in one by using the above example (`vectorXYZ_ = QVector<XYZ>(size)`).